### ISSUE TITLE
[Bug] Case cannot be accessed nor deleted

### ISSUE THREAD

**Issue by meelich** [TheHive-Project/TheHive#1638]
### Request Type
[Bug]

### Work Environment

| Question              | Answer
|---------------------------|--------------------
| OS version (server)       | Debian 10
| OS version (client)       | Windows 10
| TheHive version / git hash   | 4.0.0-1
| Package Type              | DEB 

### Problem Description
We use a modified version of the Ews2Case.py from Synapse to ingest emails and convert them into alerts and from alerts to cases. This works generally without any issues. However, we have one case (caseId: 349) that we can not access neither through the UI nor via the API. The UI won't display any other cases if case 349 would be displayed on the same page in the "list of cases"-overview. Anything works fine, if you are able to filter out case 349 (or change to another page in the overview). When you try to call it in the UI, you get this error message in the log file:
```
[ERROR] from akka.actor.ActorSystemImpl in application-akka.actor.default-dispatcher-11 [|] Response stream for [POST /api/v1/query] failed with 'Too many elements in result (2 found)'. Aborting connection.
org.thp.scalligraph.InternalError: Too many elements in result (2 found)
        at org.thp.scalligraph.steps.StepsOps$.atMostOneOf(StepsOps.scala:116)
        at org.thp.thehive.services.CaseSteps.$anonfun$richCaseWithCustomRenderer$8(CaseSrv.scala:395)
        at org.thp.scalligraph.steps.StepsOps$TraversalOps.$anonfun$map$1(StepsOps.scala:402)
        at gremlin.scala.GremlinScala.$anonfun$map$1(GremlinScala.scala:166)
        at org.apache.tinkerpop.gremlin.process.traversal.step.map.LambdaMapStep.map(LambdaMapStep.java:42)
        at org.apache.tinkerpop.gremlin.process.traversal.step.map.MapStep.processNextStart(MapStep.java:37)
        at org.apache.tinkerpop.gremlin.process.traversal.step.util.AbstractStep.hasNext(AbstractStep.java:143)
        at org.apache.tinkerpop.gremlin.process.traversal.util.DefaultTraversal.hasNext(DefaultTraversal.java:197)
        at scala.collection.convert.Wrappers$JIteratorWrapper.hasNext(Wrappers.scala:43)
        at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
        at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
        at akka.stream.impl.fusing.StatefulMapConcat$$anon$49.hasNext(Ops.scala:2176)
        at akka.stream.impl.fusing.StatefulMapConcat$$anon$49.pushPull(Ops.scala:2181)
        at akka.stream.impl.fusing.StatefulMapConcat$$anon$49.onPush(Ops.scala:2193)
        at akka.stream.impl.fusing.GraphInterpreter.processPush(GraphInterpreter.scala:541)
        at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:527)
        at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:390)
        at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:625)
        at akka.stream.impl.fusing.GraphInterpreterShell.init(ActorGraphInterpreter.scala:595)
        at akka.stream.impl.fusing.ActorGraphInterpreter.tryInit(ActorGraphInterpreter.scala:702)
        at akka.stream.impl.fusing.ActorGraphInterpreter.finishShellRegistration(ActorGraphInterpreter.scala:745)
        at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$shortCircuitBatch(ActorGraphInterpreter.scala:760)
        at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:785)
        at akka.actor.Actor.aroundReceive(Actor.scala:535)
        at akka.actor.Actor.aroundReceive$(Actor.scala:533)
        at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:691)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:575)
        at akka.actor.ActorCell.invoke(ActorCell.scala:545)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270)
        at akka.dispatch.Mailbox.run(Mailbox.scala:231)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:243)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```
When you access the case through the API, we're getting another error message:
```
curl -XPOST -H 'Content-Type: application/json' -H 'Authorization: Bearer APIKey' 'http://127.0.0.1:9000/api/case/_search' -d '{ "query": {"caseId":349}}'
curl: (56) Recv failure: Connection reset by peer
```

We also tried to delete it via API, but we're not getting any response back from TheHive (no error or confirmation). We created a workaround by adding 10 dummy cases, so case 349 would be pushed back far enough to no longer prevent other cases from being displayed.

### Steps to Reproduce
1. Getting case: curl -XPOST -H 'Content-Type: application/json' -H 'Authorization: Bearer APIKey' 'http://127.0.0.1:9000/api/case/_search' -d '{ "query": {"caseId":349}}' 
leads to error message: curl: (56) Recv failure: Connection reset by peer
2. Deleting case:  curl -XDELETE -H 'Content-Type: application/json' -H 'Authorization: Bearer APIKey' 'http://127.0.0.1:9000/api/case/349'
leads to nothing, but case has not been deleted as error still persists.

### Complementary information
Log entries where shared above, we're just interested in understanding what caused the issue (so we can prevent it from happening again), and how we can delete the faulty case to remove the issue.
Happy to share any additional information that might be required. Thanks.

**Comment by To-om** [comment]
This is fixed in 4.0.1

**Comment by meelich** [comment]
@To-om Thanks for the update. How can we remove this faulty case from TheHive now? Is there a way to kick it out of the database? It's still causing issues.
